peer/brontide.go

```diff
@@ -2431,25 +2431,29 @@ out:
 				continue
 			}
 
-			// We'll query the localChanCfg of the new channel to determine the
-			// minimum HTLC value that can be forwarded. For the maximum HTLC
-			// value that can be forwarded and fees we'll use the default
-			// values, as they currently are always set to the default values
-			// at initial channel creation. Note that the maximum HTLC value
-			// defaults to the cap on the total value of outstanding HTLCs.
-			//
-			// TODO(guggero): We should instead pass in the current
-			// forwarding policy from the funding manager to avoid
-			// needing us to update the link once the channel is
-			// announced to the network with custom user values.
-			fwdMinHtlc := lnChan.FwdMinHtlc()
-			defaultPolicy := p.cfg.RoutingPolicy
+			// We'll query the channel DB for the new channel's
+			// initial forwarding policies to determine the policy
+			// we start out with.
+			db := p.cfg.ChannelDB
```

Comment:
I think in `brontide`, we can eventually replace this call with `FetchChannelEdgesByOutpoint` to get the policy. Atm the channel is prematurely added to `brontide` by our funding manager, and I have a coming PR to fix it. Baiscally we want to only notify `brontide`(`AddNewChannel`) after both side has processed the `channel_ready` message. This way we should be able to call `FetchChannelEdgesByOutpoint` to get the policy.

In the long run, we probably won't need to care about channel policy in `brontide` at all and let `htlcswitch` fetch the policy when creating the link.

Reply:
Hmm, but won't the edge only be available in the channel graph _after_ we've announced it to the network and handed it to the router? Which I think currently happens after we've handed the channel to `brontide`. But I agree, it would be nice if we could do it that way around, I'm just not sure how much additional refactor that would currently mean. I'll look into it.

Reply:
Discussed this offline, we'll eventually want to refactor this, but not in this PR.